src/flamenco/runtime/tests/fd_exec_instr_test.c

```diff
@@ -515,11 +545,363 @@ _context_create( fd_exec_instr_test_runner_t *        runner,
+  if( !slot_ctx->sysvar_cache->has_clock ) {
+    FD_LOG_WARNING(( "Clock sysvar account not provided in account shared data" ));
+    return 0;
+  }
+
+  /* Sanity check to ensure provided slot matches clock slot */
+  if( slot_ctx->sysvar_cache->val_clock->slot != test_ctx->slot_ctx.slot ) {
+    FD_LOG_WARNING(( "Clock slot does not match provided slot ctx slot" ));
+    return 0;
+  }
+
+  /* Set slot bank variables (defaults obtained from GenesisConfig::default() in Agave) */
+  slot_ctx->slot_bank.slot                                            = test_ctx->slot_ctx.slot;
+  slot_ctx->slot_bank.prev_slot                                       = slot_ctx->slot_bank.slot - 1; // Can underflow, but its fine since it will correctly be ULONG_MAX
+  slot_ctx->slot_bank.fee_rate_governor.burn_percent                  = 50;
+  slot_ctx->slot_bank.fee_rate_governor.min_lamports_per_signature    = 0;
+  slot_ctx->slot_bank.fee_rate_governor.max_lamports_per_signature    = 0;
+  slot_ctx->slot_bank.fee_rate_governor.target_lamports_per_signature = 10000;
+  slot_ctx->slot_bank.fee_rate_governor.target_signatures_per_slot    = 20000;
+  slot_ctx->slot_bank.lamports_per_signature                          = 5000;
+  slot_ctx->prev_lamports_per_signature                               = 5000;
+
+  /* Set epoch bank variables if not present (defaults obtained from GenesisConfig::default() in Agave) */
+  fd_epoch_schedule_t default_epoch_schedule = {
+                                                .slots_per_epoch             = 432000,
+                                                .leader_schedule_slot_offset = 432000,
+                                                .warmup                      = 1,
+                                                .first_normal_epoch          = 14,
+                                                .first_normal_slot           = 524256
+                                               };
+  fd_rent_t           default_rent           = {
+                                                .lamports_per_uint8_year     = 3480,
+                                                .exemption_threshold         = 2.0,
+                                                .burn_percent                = 50
+                                               };
+  epoch_bank->epoch_schedule = default_epoch_schedule;
+  epoch_bank->rent           = default_rent;
+  epoch_bank->ticks_per_slot = 64;
+
+  // Override default values if provided
+  if( slot_ctx->sysvar_cache->has_epoch_schedule ) {
+    epoch_bank->epoch_schedule = *slot_ctx->sysvar_cache->val_epoch_schedule;
+    fd_sysvar_epoch_schedule_init( slot_ctx );
+  }
+  if( slot_ctx->sysvar_cache->has_rent ) {
+    epoch_bank->rent = *slot_ctx->sysvar_cache->val_rent;
+    fd_sysvar_rent_init( slot_ctx );
+  }
+
+  /* Restore sysvar cache (again, since we may need to provide default sysvars) */
+  fd_sysvar_cache_restore( slot_ctx->sysvar_cache, acc_mgr, funk_txn );
```

Comment:
(For future follow-up:) maybe we should make it so that if the sysvar is invalid, we _don't_ use the default value but instead fail out? Feels more intuitive and easier for people debugging to understand.